### PR TITLE
Add support for advanced menu item properties to Navigation screen

### PR DESCRIPTION
**Pull request by tellthemachines**
## Description

Navigation link blocks in Navigation screen now sync their advanced properties with their corresponding menu items.

## How has this been tested?

Add description, classname, rel or link title to Navigation link block in Navigation screen. Check that they are reflected in `nav-menus.php`. Add properties to menu item in `nav-menus.php` and check that they are reflected in Navigation screen. Also check that rel, classname and link title render correctly in the front end.

## Types of changes

New feature (non-breaking change which adds functionality)

## Checklist:
- [ ] My code is tested.
- [ ] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

**Review by kevin940726 — APPROVED** [review]
LGTM 👍

File: packages/edit-navigation/src/store/utils.js [hunk 1]
```diff
@@ -154,6 +154,10 @@ export function computeCustomizedAttribute(
 				title: block.attributes?.label,
 				original_title: '',
 				url: block.attributes.url,
+				description: block.attributes.description,
+				xfn: block.attributes.rel.split( ' ' ),
+				classes: block.attributes.className.split( ' ' ),
```
Comment (noisysocks) [inline packages/edit-navigation/src/store/utils.js:159]: This throws an error when `rel` or `className` is `undefined`. 

1. Add a new link to the Navigation screen.
2. Click _Save_.
3. Note the _There was an error._ message.

Comment (tellthemachines) [reply packages/edit-navigation/src/store/utils.js:159]: Whoops, I'll fix it!

Comment (tellthemachines) [reply packages/edit-navigation/src/store/utils.js:159]: Here: #25229

**Pull request merged**